### ISSUE TITLE
LineSeries not registering "hit" events on segments IE10/11

### ISSUE THREAD

**Issue by dswitzer** [amcharts/amcharts4#161]
I'm trying attach a hit event to `series.segments.template` and this works fine most browsers, but in Internet Explorer 10/11 nothing is triggered when I trigger a hit event. Here's the code I'm using:

```
series.segments.template.interactionsEnabled = true;
series.segments.template.events.on("hit", hitHandler);
```

In looking into the issue, when I dump out the `series.segmentsContainer.dom` element and use "Show in DOM Explorer" the IE Dev Tools it says:

> The requested element is no longer attached to the DOM

Now maybe this is a Dev Tools bug and the element does actually exist. I did try adding an ID to the element, but was never able to find in the DOM Explorer. 

I noticed the DOM structure is quite a bit different than the same chart rendered in Firefox or Chrome.

So I'm not sure if the elements are not actually added to the DOM or maybe there's a z-index issue with the elements and they're being rendered behind some other element that's stealing the mouse events.

Here's an example that shows attaching a `hit` event to the segments:

https://codepen.io/anon/pen/rrMVBy

This works fine in all modern browsers, but doesn't work at all in Internet Explorer.

**Comment by martynasma** [comment]
OK, looks like I found the culprit:

![image](https://user-images.githubusercontent.com/1458520/43020136-b87396f0-8c67-11e8-8503-464adf486e26.png)

Now, we just need to figure out where and why is it coming from.

Thanks for the heads up!

**Comment by dswitzer** [comment]
Good catch!

I'm not sure if this helps at all, but when trying to find a workaround for the issue I was messing around with attaching `hit` events to the `chart` object and various containers. What I found is that the `hit` doesn't always trigger properly. This occurs in Firefox and Chrome too.

Here's an example:

https://codepen.io/anon/pen/gjgMqM

The `hit` event should trigger no matter where I click, but some areas in the blue fill will not trigger the event. The areas where the `hit` doesn't trigger isn't intuitive, you've got to click around until you find a spot that doesn't register. There's usually a good amount of space that's a dead zone like this.

**Comment by martynasma** [comment]
Indeed. I'm suspecting to believe that this has to do with masks. Which is probably a different issue than what we're seeing with IEs.

Area before first grid line and after the last one is not clickable. It's not random.

**Comment by martynasma** [comment]
OK, so IE click issue is resolved in dev and will be in the next version we release.

Regarding other click issue. It seems you have a "hit" event set on the whole chart. There is no guarantee that the whole chart is covered with elements that can catch events, hence them not working everywhere.

If you want to have that, you can enable chart background so there's something covering whole of the area:

```
chart.background.opacity = 0.001
```

**Comment by martynasma** [comment]
Fixed in [4.0.0-beta.35](https://github.com/amcharts/amcharts4/releases/tag/4.0.0-beta.35).

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by dswitzer** [comment]
I wanted to confirm this has been resolved.

Great work!
